source/signaling_api.c

```diff
@@ -0,0 +1,1447 @@
+                } else {
+                    /* Skip unknown messages. */
+                }
+
+                jsonResult = JSON_Iterate( pIceSingleServerBuffer, iceSingleServerBufferLength, &iceSingleServerStart, &iceSingleServerNext, &pair );
+            }
+
+        } else {
+            /* All parsed. */
+            pIceServerList->iceServerNum++;
+            break;
+        }
+    }
+
+    return result;
+}
+
+SignalingResult_t Signaling_Init( SignalingContext_t *pCtx, SignalingAwsControlPlaneInfo_t *pAwsControlPlaneInfo )
+{
+    SignalingResult_t result = SIGNALING_RESULT_OK;
+    int length = 0;
+
+    /* input check */
+    if (pCtx == NULL || pAwsControlPlaneInfo == NULL) {
+        result = SIGNALING_RESULT_BAD_PARAM;
+    }
+
+    /* Set region. */
+    if (result == SIGNALING_RESULT_OK) {
+        memset(pCtx, 0, sizeof(SignalingContext_t));
+
+        if (pAwsControlPlaneInfo->pRegion == NULL) {
+            length = snprintf(pCtx->region, SIGNALING_AWS_REGION_MAX_LENGTH, "%.*s",
+                              (int) strlen(SIGNALING_AWS_DEFAULT_REGION), SIGNALING_AWS_DEFAULT_REGION);
+        } else {
+            length = snprintf(pCtx->region, SIGNALING_AWS_REGION_MAX_LENGTH, "%.*s",
+                              (int) pAwsControlPlaneInfo->regionLength, pAwsControlPlaneInfo->pRegion);
+        }
+
+        if (length < 0) { //LCOV_EXCL_BR_LINE
+            result = SIGNALING_RESULT_SNPRINTF_ERROR; // LCOV_EXCL_LINE
+        }
+        else if (length >= SIGNALING_AWS_CONTROL_PLANE_URL_MAX_LENGTH) {
+            result = SIGNALING_RESULT_REGION_LENGTH_TOO_LARGE;
+        }
+        else {
+            pCtx->regionLength = length;
+        }
+    }
+
+    /* Set control plane. */
+    if (result == SIGNALING_RESULT_OK) {
+        if (pAwsControlPlaneInfo->pControlPlaneUrl == NULL) {
+            length = snprintf(pCtx->controlPlaneUrl, SIGNALING_AWS_CONTROL_PLANE_URL_MAX_LENGTH, "%s%s.%s%s",
+                              AWS_CONTROL_PLANE_URI_PREFIX, AWS_KINESIS_VIDEO_SERVICE_NAME, 
+                              pCtx->region, AWS_CONTROL_PLANE_URI_POSTFIX);
+        } else {
+            length = snprintf(pCtx->controlPlaneUrl, SIGNALING_AWS_CONTROL_PLANE_URL_MAX_LENGTH, "%.*s",
+                              (int) pAwsControlPlaneInfo->controlPlaneUrlLength, pAwsControlPlaneInfo->pControlPlaneUrl);
+        }
+
+        if (length < 0) { //LCOV_EXCL_BR_LINE
+            result = SIGNALING_RESULT_SNPRINTF_ERROR; // LCOV_EXCL_LINE
+        }
+        else if (length >= SIGNALING_AWS_CONTROL_PLANE_URL_MAX_LENGTH) {
+            result = SIGNALING_RESULT_OUT_OF_MEMORY;
+        }
+        else {
+            pCtx->controlPlaneUrlLength = length;
+        }
+    }
+
+    return result;
+}
+
+SignalingResult_t Signaling_constructDescribeSignalingChannelRequest( SignalingContext_t *pCtx, SignalingDescribeSignalingChannelRequest_t * pDescribeSignalingChannelRequest, SignalingRequest_t *pRequestBuffer )

```

Comment:
Signaling_constructDescribeSignalingChannelRequest -> Signaling_ConstructDescribeSignalingChannelRequest
C should be capital.